File: source/SharpFlame/Tools.cs
namespace SharpFlame
{
	public sealed class modTools
	{
		
		public struct sEachTool
		{
			public clsTool ObjectSelect;
			public clsTool TextureBrush;
			public clsTool TerrainBrush;
			public clsTool TerrainFill;
			public clsTool RoadPlace;
			public clsTool RoadLines;
			public clsTool RoadRemove;
			public clsTool CliffTriangle;
			public clsTool CliffBrush;
			public clsTool CliffRemove;
			public clsTool HeightSetBrush;
			public clsTool HeightChangeBrush;
			public clsTool HeightSmoothBrush;
			public clsTool ObjectPlace;
			public clsTool ObjectLines;
			public clsTool TerrainSelect;
			public clsTool Gateways;
		}
		public static sEachTool Tools;
		
		public static void CreateTools()
		{
			
			clsTool newTool = default(clsTool);
			
			newTool = new clsTool();
			Tools.ObjectSelect = newTool;
			
			newTool = new clsTool();
			Tools.TextureBrush = newTool;
			
			newTool = new clsTool();
			Tools.TerrainBrush = newTool;
			
			newTool = new clsTool();
			Tools.TerrainFill = newTool;
			
			newTool = new clsTool();
			Tools.RoadPlace = newTool;
			
			newTool = new clsTool();
			Tools.RoadLines = newTool;
			
			newTool = new clsTool();
			Tools.RoadRemove = newTool;
			
			newTool = new clsTool();
			Tools.CliffTriangle = newTool;
			
			newTool = new clsTool();
			Tools.CliffBrush = newTool;
			
			newTool = new clsTool();
			Tools.CliffRemove = newTool;
			
			newTool = new clsTool();
			Tools.HeightSetBrush = newTool;
			
			newTool = new clsTool();
			Tools.HeightChangeBrush = newTool;
			
			newTool = new clsTool();
			Tools.HeightSmoothBrush = newTool;
			
			newTool = new clsTool();
			Tools.ObjectPlace = newTool;
			
			newTool = new clsTool();
			Tools.ObjectLines = newTool;
			
			newTool = new clsTool();
			Tools.TerrainSelect = newTool;
			
			newTool = new clsTool();
			Tools.Gateways = newTool;
			
			_Tool = Tools.TextureBrush;
			_PreviousTool = _Tool;
		}
		
		private static clsTool _Tool;
public static clsTool Tool
		{
			get
			{
				return _Tool;
			}
			set
			{
				_PreviousTool = _Tool;
				_Tool = value;
			}
		}
		private static clsTool _PreviousTool;
public static clsTool PreviousTool
		{
			get
			{
				return _PreviousTool;
			}
		}
	}
	
	public class clsTool
	{
		
	}
	
}

File: reference/FlaMEVB/Tools.vb

Public Module modTools

    Public Structure sEachTool
        Public ObjectSelect As clsTool
        Public TextureBrush As clsTool
        Public TerrainBrush As clsTool
        Public TerrainFill As clsTool
        Public RoadPlace As clsTool
        Public RoadLines As clsTool
        Public RoadRemove As clsTool
        Public CliffTriangle As clsTool
        Public CliffBrush As clsTool
        Public CliffRemove As clsTool
        Public HeightSetBrush As clsTool
        Public HeightChangeBrush As clsTool
        Public HeightSmoothBrush As clsTool
        Public ObjectPlace As clsTool
        Public ObjectLines As clsTool
        Public TerrainSelect As clsTool
        Public Gateways As clsTool
    End Structure
    Public Tools As sEachTool

    Public Sub CreateTools()

        Dim newTool As clsTool

        newTool = New clsTool
        Tools.ObjectSelect = newTool

        newTool = New clsTool
        Tools.TextureBrush = newTool

        newTool = New clsTool
        Tools.TerrainBrush = newTool

        newTool = New clsTool
        Tools.TerrainFill = newTool

        newTool = New clsTool
        Tools.RoadPlace = newTool

        newTool = New clsTool
        Tools.RoadLines = newTool

        newTool = New clsTool
        Tools.RoadRemove = newTool

        newTool = New clsTool
        Tools.CliffTriangle = newTool

        newTool = New clsTool
        Tools.CliffBrush = newTool

        newTool = New clsTool
        Tools.CliffRemove = newTool

        newTool = New clsTool
        Tools.HeightSetBrush = newTool

        newTool = New clsTool
        Tools.HeightChangeBrush = newTool

        newTool = New clsTool
        Tools.HeightSmoothBrush = newTool

        newTool = New clsTool
        Tools.ObjectPlace = newTool

        newTool = New clsTool
        Tools.ObjectLines = newTool

        newTool = New clsTool
        Tools.TerrainSelect = newTool

        newTool = New clsTool
        Tools.Gateways = newTool

        _Tool = Tools.TextureBrush
        _PreviousTool = _Tool
    End Sub

    Private _Tool As clsTool
    Public Property Tool As clsTool
        Get
            Return _Tool
        End Get
        Set(value As clsTool)
            _PreviousTool = _Tool
            _Tool = value
        End Set
    End Property
    Private _PreviousTool As clsTool
    Public ReadOnly Property PreviousTool As clsTool
        Get
            Return _PreviousTool
        End Get
    End Property
End Module

Public Class clsTool

End Class
